python/cudf/cudf/tests/test_parquet.py

```diff
@@ -901,6 +901,55 @@ def test_parquet_reader_list_large_mixed(tmpdir):
     assert_eq(expect, got, check_dtype=False)
 
 
+def test_parquet_reader_list_large_multi_rowgroup(tmpdir):
+    # > 800 row groups
+    num_rows = 2_000_000
+    num_docs = num_rows / 2
+    num_categories = 1_000
+    row_group_size = 1000
+
+    # generate a random pairing of doc: category
+    documents = cudf.DataFrame(
+        {
+            "document_id": np.random.randint(num_docs, size=num_rows),
+            "category_id": np.random.randint(num_categories, size=num_rows),
```

Comment:
```suggestion
            "document_id": cupy.random.randint(num_docs, size=num_rows),
            "category_id": cupy.random.randint(num_categories, size=num_rows),
```

I think we could use cupy equivalent here.

note: we'll have to add a `cupy` import at top

Reply:
That's actually what the initial test was, I just wasn't sure if it was cool to add cupy like that.

Reply:
yes, we can use the cupy API to avoid host to device delays in copying.

We can do `cupy.random.seed(0)` in this test as we do here:

https://github.com/rapidsai/cudf/blob/7c055c1a6618364d39b5dea5773c206f96b67a47/python/dask_cudf/dask_cudf/tests/test_join.py#L20-L23

Reply:
Done.